lib/Sema/TypeCheckType.cpp

```diff
@@ -830,6 +833,41 @@ static Type applyGenericArguments(Type type, TypeResolution resolution,
       diags.diagnose(loc, diag::use_of_void_pointer, "").
         fixItReplace(generic->getSourceRange(), "UnsafeRawPointer");
   }
+
+  if (auto clangDecl = decl->getClangDecl()) {
+    if (auto classTemplateDecl =
+            dyn_cast<clang::ClassTemplateDecl>(clangDecl)) {
+      SmallVector<clang::TemplateArgument, 2> templateArguments;
```

Comment:
When you want to support non-clang types, it shouldn't be too hard to replace this with `ASTContext ::getClangTemplateArguments`.

Reply:
Then we'll have the same issue with Swift generic types, though. Anyway, that's a problem for later 😅

Reply:
Yup I'll definitely use `getClangTemplateArguments` once it lands.

Reply:
`getClangTemplateArguments` is implemented in master, if you want to use it in this patch. Otherwise, it could certainly be a follow up.

Reply:
Done, thanks for reminding me.